itests/framework/framework.go

```diff
@@ -758,15 +762,17 @@ func (f *TestFramework) ExtractFileFromCAR(ctx context.Context, t *testing.T, fi
 	return tmpFile
 }
 
-func (f *TestFramework) RetrieveDirect(ctx context.Context, t *testing.T, root cid.Cid, pieceCid *cid.Cid, carExport bool) string {
+func (f *TestFramework) RetrieveDirect(ctx context.Context, t *testing.T, root cid.Cid, pieceCid *cid.Cid, carExport bool, extractCar bool, selectorNode datamodel.Node) string {
 	offer, err := f.FullNode.ClientMinerQueryOffer(ctx, f.MinerAddr, root, pieceCid)
 	require.NoError(t, err)
 
-	return f.retrieve(ctx, t, offer, carExport)
+	return f.retrieve(ctx, t, offer, carExport, extractCar, selectorNode)
 }
 
-func (f *TestFramework) retrieve(ctx context.Context, t *testing.T, offer lapi.QueryOffer, carExport bool) string {
+func (f *TestFramework) retrieve(ctx context.Context, t *testing.T, offer lapi.QueryOffer, carExport bool, extractCar bool, selectorNode datamodel.Node) string {
```

Comment:
Both variables `carExport` and `extractCar` seems to be doing the same thing.

Reply:
Sadly no, `carExport` gets sent along to the Lotus API to ask for a CAR (although I'm not sure what `IsCar: false` is going to do, perhaps a piece fetch?), while `extractCar` is used to determine whether to unpack the CAR into a unixfs file. I wish they could be a single boolean because I'm not happy extending this function signature even more.

The alternative might be to just expose a `RetrieveCar` method that handles this internally with `retrieve()`, that might make it a bit cleaner.

Reply:
Actually .. it looks like this argument isn't actually used by anything else, perhaps this was originally copied out of lotus and we only ever ask for CAR exports.

So I've just removed that argument and replaced it with my new one that determines whether to extract the CAR or not; hard-wiring a `true` to the lotus call.